File: Java/Easy/MaximumDepthofBinaryTree.java
package easy;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class MaximumDepthofBinaryTree {

    public int maxDepth(TreeNode root) {
      if(root == null) return 0;
        int left = maxDepth(root.left);
        int right = maxDepth(root.right);
        
        return Math.max(left, right)+1;
    }
}

/* 
class PastSolution { 
public:
    int maxDepth(TreeNode* root) {
        if(root == nullptr){
            return 0;
        }
        int l = maxDepth(root->left);
        int r = maxDepth(root->right);
        if (l>r){
            return l+1;
        }
        else{
            return r+1;
        }
    }
};
 */

File: C#/Easy/Maximum Depth of Binary Tree.cs

  //Definition for a binary tree node.
  public class TreeNode {
      public int val;
      public TreeNode left;
      public TreeNode right;
      public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
          this.val = val;
          this.left = left;
          this.right = right;
      }
  }
 
public class Solution
{
    public int MaxDepth(TreeNode root)
    {
        return helper(root, 0);
    }
    private int helper(TreeNode root, int depth)
    {
        if (root == null)
        {
            return depth;
        }
        return Math.Max(helper(root.left, depth + 1), helper(root.right, depth + 1));
    }
}